### PR TITLE
migrate: plugins/histogram/polymer3

### PR DESCRIPTION
**Pull request by wchargin**
This patch covers `vz_histogram_timeseries` and `tf_histogram_dashboard`.

Test Plan:
With `TB_POLYMER=1 bazel run //tensorboard`,

  - histograms render;
  - the histogram mode (overlay/offset) selector works;
  - the offset time axis works;
  - toggling runs in the run selector works;
  - entering a regex filter in the run selector works;
  - entering a regex tag filter works;
  - adjusting the pagination limit works.

### CODE REVIEW

File: tensorboard/plugins/histogram/polymer3/tf_histogram_dashboard/tf-histogram-core.ts [hunk 1]
```diff
@@ -13,8 +11,4 @@
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
--->
-
-<link rel="import" href="../tf-imports/d3.html" />
-
-<script src="histogramCore.js"></script>
+==============================================================================*/
```
Comment (psybuzz) [inline tensorboard/plugins/histogram/polymer3/tf_histogram_dashboard/tf-histogram-core.ts:14]: Can this unused file can be deleted?

File: tensorboard/plugins/histogram/polymer3/tf_histogram_dashboard/tf-histogram-dashboard.html [hunk 2]
```diff
@@ -1,252 +0,0 @@
...
-        _dataNotFound: Boolean,
-        _tagFilter: String,
-        _restamp: {
-          type: Boolean,
-          value: false,
-        },
-
-        // Categories must only be computed after _dataNotFound is found to be
-        // true and then polymer DOM templating responds to that finding. We
-        // thus use this property to guard when categories are computed.
```
Comment (psybuzz) [inline tensorboard/plugins/histogram/polymer3/tf_histogram_dashboard/tf-histogram-dashboard.html:177]: Preserve this comment?

File: tensorboard/plugins/histogram/polymer3/tf_histogram_dashboard/tf-histogram-dashboard.html [hunk 3]
```diff
@@ -1,252 +0,0 @@
...
-        },
-
-        _requestManager: {
-          type: Object,
-          value: () => new tf_backend.RequestManager(),
-        },
-      },
-
-      listeners: {
-        'content-visibility-changed': '_redrawCategoryPane',
```
Comment (psybuzz) [inline tensorboard/plugins/histogram/polymer3/tf_histogram_dashboard/tf-histogram-dashboard.html:192]: The auto-migration script doesn't cover `listeners`, so we should add something like

`this.addEventListener('content-visibility-changed', this._redrawCategoryPane.bind(this))`
in either the `ready()` or `connectedCallback()`

**Comment by psybuzz** [comment]
Tested it out with the commits mentioned, and it looks great!

File: tensorboard/plugins/histogram/polymer3/tf_histogram_dashboard/tf-histogram-dashboard.html (hunk 2 above)
Comment (wchargin) [reply tensorboard/plugins/histogram/polymer3/tf_histogram_dashboard/tf-histogram-dashboard.html:177]: Done.

File: tensorboard/plugins/histogram/polymer3/tf_histogram_dashboard/tf-histogram-dashboard.html (hunk 3 above)
Comment (wchargin) [reply tensorboard/plugins/histogram/polymer3/tf_histogram_dashboard/tf-histogram-dashboard.html:192]: We looked into this a bit. This isn’t a standard DOM event; rather, it’s
a custom Polymer event that was originally added in #1465. But there
isn’t any code to actually fire the event; `git log -S` shows that it
was removed in #2273 via #2392. Thus, while the absence of this listener
may be a bug, it’s not new, so I’ll keep this migration patch as is to
make things simple. Thanks for the keen eye, though!

**Comment by wchargin** [comment]
Test sync; <http://cl/325558004>.

**Pull request merged**